core/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -175,6 +206,16 @@ public String concreteSingleIndex(ClusterState state, IndicesRequest request) {
     }
 
     /**
+     * @return whether the specified alias or index exists. If the alias or index contains datemath then that is resolved too.
+     */
+    public boolean hasIndexOrAliasAndResolveDateMath(String aliasOrIndex, ClusterState state) {
+        assert expressionResolvers.get(0) instanceof DateMathExpressionResolver;
```

Comment:
given that we add the expression resolvers manually to the list, maybe keep a reference to it instead so we can use it directly instead of getting it from the list?